src/main/resources/com/google/api/codegen/java/http_request.snip

```diff
@@ -0,0 +1,233 @@
+@extends "java/common.snip"
+@extends "java/method_sample.snip"
+
+@snippet generate(apiFile)
+  {@renderFileHeader(apiFile.fileHeader)}
+
+  @@Generated("by GAPIC")
+  @@BetaApi
+  {@requestClassDef(apiFile.request)}
+@end
+
+@private requestClassDef(request)
+  public final class {@request.typeName} implements Message {
```

Comment:
Done. Request and non-request messages share the same template and ViewModel. However, non-request message types won't implement ApiMessage because the only planned method in ApiMessage is populateMapMethod, and that is undefined for non-request message types.